src/frontend/http.rs

```diff
@@ -100,6 +102,23 @@ fn plan_to_etag(plan: &LogicalPlan) -> String {
     encode(hasher.finalize())
 }
 
+// Construct a content-type header value that also includes schema information.
+fn content_type_with_schema(schema: SchemaRef) -> HeaderValue {
+    let schema_string = schema_to_json(schema.as_ref()).to_string();
+    let output = utf8_percent_encode(&schema_string, NON_ALPHANUMERIC);
+
+    match HeaderValue::from_str(
+        format!("application/json; arrow-schema-escaped={output}").as_str(),
```

Comment:
I think we should still call it `arrow-schema`, even though it's percent encoded.

Reply:
That's fine with me. @milesrichardson just an fyi that I'm revising the parameter name back to `arrow-schema`.